test/rpc/test_rpc_transactions.py

```diff
@@ -431,12 +443,96 @@ def test_add_invoke_transaction():
         private_key=PREDEPLOYED_ACCOUNT_PRIVATE_KEY,
         nonce=0,
         version=SUPPORTED_RPC_TX_VERSION,
-        max_fee=0,
+        max_fee=max_fee,
     )
 
     invoke_transaction = RpcBroadcastedInvokeTxnV1(
         type="INVOKE",
-        max_fee=rpc_felt(0),
+        max_fee=rpc_felt(max_fee),
+        version=hex(SUPPORTED_RPC_TX_VERSION),
+        signature=[rpc_felt(sig) for sig in signature],
+        nonce=rpc_felt(get_nonce(PREDEPLOYED_ACCOUNT_ADDRESS)),
+        sender_address=rpc_felt(PREDEPLOYED_ACCOUNT_ADDRESS),
+        calldata=[rpc_felt(data) for data in execute_calldata],
+    )
+
+    resp = rpc_call(
+        "starknet_addInvokeTransaction",
+        params={"invoke_transaction": invoke_transaction},
+    )
+    receipt = resp["result"]
+
+    storage = gateway_call(
+        "get_storage_at",
+        contractAddress=contract_address,
+        key=get_storage_var_address("balance"),
+    )
+
+    assert set(receipt.keys()) == {"transaction_hash"}
+    assert is_felt(receipt["transaction_hash"])
+    assert storage == hex(initial_balance + amount1 + amount2)
+
+
+@pytest.mark.usefixtures("devnet_with_account")
+def test_add_invoke_transaction_with_max_fee_0():
```

Comment:
So far we've kept `test/rpc` as a place for JSON-RPC-specific tests. Actual features have been tested in other parts of `test/`. A separate directory with tests not relying on JSON-RPC would be better IMO.

Reply:
I'm not sure to understand what you mean here.
I have updated all tests using `declare` transaction since max_fee==0 is no more supported.

Reply:
The logic of the two tests which you contributed is good, but...

I meant that in Devnet, JSON-RPC support has been treated as a separate feature. And the treatment of txs with max_fee=0 should not be tested using this separate feature (JSON-RPC) and in an existing test file (`test/rpc/test_rpc_transactions.py`).

That's why I recommended testing without JSON-RPC (e.g. the way it's done in `test/test_account.py`) and in a separate file (I wrote "separate directory" in the previous comment but I actually meant a separate file, e.g. `test/test_max_fee_0.py`). The new file would then contain not just a test for invoke with max_fee=0 but also other transactions.

Feel free to contact me on Telegram if something is not clear.

Reply:
Updated PR to move test in a separate file.